### PR TITLE
Migrate deployment tasks to CircleCI

### LINKED ISSUE #3266: CircleCI build deployment
## Abstract

Previously, TravisCI was used to perform nightly and release builds which are then deployed on github & the AWS nightly bucket. Since the move to CircleCI, these haven't been ported along with the rest of the CI tasks. 

## Specification

1. A nightly build should be triggered on TravisCI and the build artifacts should be deployed on the nightly AWS bucket.
2. Tagging a release on github should trigger a build whose artifacts are deployed on github for both MacOS and Linux.

Parent issue #3072

### PR DESCRIPTION
**Pull request by jomuel**
The complete and tested workflows for nightly builds and releases on CircleCI have already been merged. This PR will

- remove the Travis jobs for nightly builds and releases and
- change the deployment target for CircleCI nightly builds so they replace the Travis builds

The releases via CircleCI will then be triggered by adding `GITHUB_TOKEN` to the CircleCI context (see `.circleci/config.yml:388`.

**Edit:** The CircleCI release workflow is still missing MacOS releases. So the PR should not be merged and releases via Travis kept until that is fixed.

Closes #3266.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -375,7 +375,7 @@ jobs:
       - config-path
       - run:
           name: upload-to-digitalocean
-          command: .travis/upload_to_s3.sh $DO_S3_ACCESS_KEY $DO_S3_SECRET_KEY build/archive/ raiden-ci-test ams3.digitaloceanspaces.com
+          command: .circleci/upload_to_s3.sh $DO_S3_ACCESS_KEY $DO_S3_SECRET_KEY build/archive/ $DO_S3_SUB_SUB_DOMAIN ams3.digitaloceanspaces.com
```
Comment (rakanalh) [inline .circleci/config.yml]: Out of curiosity, where are those values defined? `DO_S3_SUB_SUB_DOMAIN` and the others

File: .travis.yml [hunk 2]
```diff
@@ -171,106 +163,3 @@ jobs:
         - TEST='raiden/tests/integration/cli'
         - TRANSPORT_OPTIONS='--transport=matrix'
         - RUN_SYNAPSE=1
-
-    - stage: deploy
```
Comment (rakanalh) [inline .travis.yml]: This means that we will not have build deployment on neither travis nor circle ci until the builld is implemented on circleCI right? if so, how soon can we get the builds up and running on circle CI? i imagine a release is going to be made during this week.

File: .circleci/config.yml (hunk 1 above)
Comment (jomuel) [reply .circleci/config.yml]: We defined a context with all these environment variables on CircleCI. So in particular the passwords/access tokens do not need to be present in the repo at all, also not in encrypted form as it was with travis.

File: .travis.yml (hunk 2 above)
Comment (jomuel) [reply .travis.yml]: Releases via CircleCI are already implemented, just skipped until the github token is added to the context. (MacOS binaries are still missing on CircleCI though.)

**Comment by LefterisJP** [comment]
>MacOS binaries are still missing on CircleCI though

So how are we going to release if 1/2 binaries can't be made with CircleCI?

**Comment by jomuel** [comment]
@lefterisjp Ulo is working on the MacOS part. Depending on when it's done, it will either become part of this PR or we will keep the travis job for MacOS releases for the next release.

**Comment by ulope** [comment]
macOS deployment added

File: .circleci/config.yml [hunk 3]
```diff
@@ -28,8 +28,17 @@ templates:
       default: ""
       type: string
 
+  default-environment: &default-environment
+    GETH_URL_LINUX: 'https://gethstore.blob.core.windows.net/builds/geth-linux-amd64-1.8.20-24d727b6.tar.gz'
+    GETH_URL_MACOS: 'https://gethstore.blob.core.windows.net/builds/geth-darwin-amd64-1.8.20-24d727b6.tar.gz'
+    GETH_VERSION: '1.8.20'
+    SOLC_URL_LINUX: 'https://github.com/ethereum/solidity/releases/download/v0.4.23/solc-static-linux'
+    SOLC_URL_MACOS: 'https://www.dropbox.com/s/4amq3on2ds1dq36/solc_0.4.23?dl=0'
```
Comment (rakanalh) [inline .circleci/config.yml]: Is this an official link for Solc MacOS?

Comment (palango) [reply .circleci/config.yml]: No, compiled by me ;)

Comment (ulope) [reply .circleci/config.yml]: No, this is a binary one of us (in this case @palango IIRC) compiled and uploaded. 
So far there are unfortunately [no official solidity macOS binaries](https://github.com/ethereum/solidity/issues/3168)

Although now with circle we could probably switch back to compiling it ourselves and storing the result in the cache.

**Pull request merged**